plugins/module_utils/crypto/cryptography_openssh.py

```diff
@@ -0,0 +1,484 @@
+                # Public exponent should always be 65537 to prevent issues
+                # if improper padding is used during signing
+                public_exponent=65537,
+                key_size=self.__size,
+                backend=backend,
+            )
+
+        def __generate_dsa_key(self):
+            return dsa.generate_private_key(
+                key_size=self.__size,
+                backend=backend,
+            )
+
+        def __generate_ed25519_key(self):
+            return Ed25519PrivateKey.generate()
+
+        def __generate_ecdsa_key(self):
+            return ec.generate_private_key(
+                self.__algorithm_parameters['ecdsa']['curves'][self.__size],
+                backend=backend,
+            )
+
+        def __load_privatekey(self, path, passphrase):
+            # OpenSSH formatted private keys are not available in Cryptography <3.0
+            try:
+                privatekey_loader = serialization.load_ssh_private_key
+            except AttributeError:
+                privatekey_loader = serialization.load_pem_private_key
+
+            if passphrase:
+                passphrase = passphrase.encode(encoding=_TEXT_ENCODING)
+
+            try:
+                with open(path, 'rb') as f:
+                    content = f.read()
+
+                    privatekey = privatekey_loader(
+                        data=content,
+                        password=passphrase,
+                        backend=backend,
+                    )
+
+                    if isinstance(privatekey, rsa.RSAPrivateKey):
+                        self.__keytype = 'rsa'
+                    elif isinstance(privatekey, dsa.DSAPrivateKey):
+                        self.__keytype = 'dsa'
+                    elif isinstance(privatekey, ec.EllipticCurvePrivateKey):
+                        self.__keytype = 'ecdsa'
+                    elif isinstance(privatekey, Ed25519PrivateKey):
+                        self.__keytype = 'ed25519'
+            except ValueError as e:
+                raise InvalidKeyFileError(e)
+            except TypeError as e:
+                raise InvalidPassphraseError(e)
+            except UnsupportedAlgorithm as e:
+                raise InvalidAlgorithmError(e)
+
+            return privatekey
+
+        def __load_publickey(self, path):
+            try:
+                with open(path + '.pub', 'rb') as f:
+                    content = f.read()
+
+                    publickey = serialization.load_ssh_public_key(
+                        data=content,
+                        backend=backend,
+                    )
+
+                    pubkey_text = content.decode(encoding=_TEXT_ENCODING)
+
+            except ValueError as e:
+                raise InvalidKeyFileError(e)
+            except UnsupportedAlgorithm as e:
+                raise InvalidAlgorithmError(e)
+
+            return publickey
+
+        def __validate_passphrase(self, passphrase):
```

Comment:
Why not make this function raise `InvalidPassphraseError` directly? That would simplify its usage.